File: utils_analytics/src/main/java/org/amoustakos/utils/analytics/EventLogger.java
package org.amoustakos.utils.analytics;

import org.amoustakos.utils.analytics.EventAction;

import java.util.Map;


/**
 * @param <T> The parameters type to be used for logging
 */
public interface EventLogger<T> {

	String SPACE = " ";

	void logEvent(
			String eventName,
			@EventAction String action,
			Map<String, String> parameters
	);

	T transformParameters(Map<String, String> attributes);

	String makeEventName(String eventName, @EventAction String action);
}

File: utils/utils_analytics/src/main/java/org/amoustakos/utils/analytics/EventLogger.kt
package org.amoustakos.utils.analytics


interface EventLogger<out T> {

    fun logEvent(
            eventName: String,
            @EventAction action: String,
            parameters: Map<String, String>
    )

    fun transformParameters(attributes: Map<String, String>): T

    fun makeEventName(eventName: String, @EventAction action: String?): String

    companion object {
        const val SPACE = " "
        const val UNDERSCORE = "_"
    }
}
